shared/common-adapters/header-hoc.desktop.js

```diff
@@ -8,13 +8,18 @@ import {globalStyles, globalColors, globalMargins} from '../styles'
 import type {Props} from './header-hoc'
 
 function HeaderHoc<P> (WrappedComponent: ReactClass<P>) {
-  return ({onBack, title, ...restProps}: Props & P) => (
-    <Box style={containerStyle}>
-      <Box style={headerStyle}>
-        <Box style={backButtonStyle}>
-          {onBack && <BackButton iconStyle={backButtonIconStyle} onClick={onBack} />}
-        </Box>
-        <Text type='Header' style={headerText}>{title}</Text>
+  return ({onBack, onCancel, headerStyle, title, ...restProps}: Props & P) => (
+    <Box style={_containerStyle}>
+      <Box style={{..._headerStyle, ...headerStyle}}>
+        {onCancel &&
+        <Box style={_cancelStyle}>
+          <Text type='BodyBigLink' onClick={onCancel}>Cancel</Text>
+        </Box>}
+        {onBack &&
+        <Box style={_backButtonStyle}>
```

Comment:
are you sure the header text is centered with this change?

I kept the back button width present so the header's margin takes it into account and is properly centered.